src/v/cluster/tm_stm.cc

```diff
@@ -116,16 +116,7 @@ tm_stm::try_change_status(
     }
     auto tx = ptx->second;
     tx.status = status;
-    auto lock_it = _tx_locks.find(tx_id);
-    vassert(
-      lock_it != _tx_locks.end(),
-      "Existence of _tx_table[{}] must emply existence of _tx_locks[{}]",
-      tx_id,
-      tx_id);
-
-    // hack: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=95599
-    auto func = [this, term, tx]() { return save_tx(term, tx); };
-    co_return co_await lock_it->second->with(func);
+    co_return co_await save_tx(term, tx);
```

Comment:
should this patch be sequenced later in the series? it didn't look like the previous patch altered save_tx (or code paths descending from save_tx) so I'm wondering if this change relies on changes later in the series?

Reply:
it's an effect of the previous commit ("tx coordinator: use locks instead of OCC") once we put the whole tx operation under a lock the intermediate locks (`_tx_locks`) became useless.